figures/compat.py

```diff
@@ -47,8 +47,8 @@ def course_grade(learner, course):
     else:  # Assume Hawthorn or greater
         course_grade = CourseGradeFactory().read(learner, course)
 
-    persistent_course_grade = PersistentCourseGrade.objects.filter(user_id=learner.id, course_id=course.id).first()
-    course_grade.passed_timestamp = persistent_course_grade.passed_timestamp if persistent_course_grade else None
+    persistent_course_grade = PersistentCourseGrade.objects.filter(user_id=learner.id, course_id=course.id).order_by('-modified').first()
+    course_grade.passed_timestamp = persistent_course_grade.passed_timestamp if persistent_course_grade and course_grade.percent == 1 else None
```

Comment:
@iam-mhaseeb `precent` grade can be less than 1 to pass a course. i.e if pass criteria is 40% then that would be 0.4

Reply:
@ziafazal That's true, I can expect multiple cases in letter_grade too. I think ideally we won't need to check `course_grade.percent` == something if persistent_course_grade populates with passed_timestamp only when a learner is actually passed. Let me verify!

Reply:
We won't need this check. persistent_course_grade populates passed_timestamp only when a learner actually pass.